source/features/ci-link.tsx

```diff
@@ -1,39 +1,80 @@
 import './ci-link.css';
 import select from 'select-dom';
-import onetime from 'onetime';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 import fetchDom from '../helpers/fetch-dom';
-import {buildRepoURL} from '../github-helpers';
+import {onCiIconLoad} from '../github-events/on-fragment-load';
+import getDefaultBranch from '../github-helpers/get-default-branch';
+import {buildRepoURL, getCurrentCommittish} from '../github-helpers';
 
 // Look for the CI icon in the latest 2 days of commits #2990
-const getIcon = onetime(async () => fetchDom(
-	buildRepoURL('commits'), [
-		'.TimelineItem--condensed:nth-of-type(-n+2) .commit-build-statuses', // TODO[2022-04-29]: GHE #4294
-		'.TimelineItem--condensed:nth-of-type(-n+2) batch-deferred-content[data-url$="checks-statuses-rollups"]',
-	].join(','),
-));
+const iconSelector = [
+	'.TimelineItem--condensed:nth-of-type(-n+2) .commit-build-statuses', // TODO[2022-04-29]: GHE #4294
+	'.TimelineItem--condensed:nth-of-type(-n+2) batch-deferred-content[data-url$="checks-statuses-rollups"]',
+].join(',');
 
-async function init(): Promise<false | void> {
-	const icon = await getIcon();
-	if (!icon) {
-		return false;
+async function getIcon(): Promise<HTMLElement | undefined> {
+	if (pageDetect.isRepoCommitList() && getCurrentCommittish() === await getDefaultBranch()) {
+		return select(iconSelector)!.cloneNode(true);
 	}
 
+	const dom = await fetchDom(buildRepoURL('commits'));
+	const icon = select(iconSelector, dom);
+	if (icon && (pageDetect.isDiscussion() || pageDetect.isDiscussionList())) {
+		const style = select('link[href*="/assets/github-"]', dom)!;
+		document.head.append(style); // #5283
+	}
+
+	return icon;
+}
+
+function appendIconToRepoHeader(icon: HTMLElement): void {
 	// Append to title (aware of forks and private repos)
 	const repoNameHeader = select('[itemprop="name"]')!.parentElement!;
 	repoNameHeader.append(icon);
 	repoNameHeader.classList.add('rgh-ci-link');
 }
 
+async function init(): Promise<false | void> {
+	const icon = await getIcon();
+	if (!icon) {
+		return false;
+	}
+
+	appendIconToRepoHeader(icon);
+}
+
+function initRepoHome(): void | false {
+	const icon = select('.file-navigation + .Box .commit-build-statuses');
+	if (!icon) {
+		return false;
+	}
```

Comment:
I think it makes sense to have a separate init for the repo homepage, since the logic and selector are different.

That said, I wonder if we could just clone the `include-fragment` and let GitHub do the rest?

Reply:
> That said, I wonder if we could just clone the include-fragment and let GitHub do the rest?

Apparently yes

Reply:
Are there any other features that can make use of this trick?

Reply:
> Are there any other features that can make use of this trick?

I wonder, but I doubt it. Usually we need to select/modify something in the `include-fragment`, so we have no choice but to wait for it to load.